src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -103,6 +103,28 @@ Copyright (c) .NET Foundation. All rights reserved.
   </ItemGroup>
 
   <!--
+    This will prevent RAR from spending time locating dependencies and related files for assemblies
+    that came from packages. PackageReference should already be promoted to a closure of Reference
+    items and we are responsible for adding package relates files to CopyLocal items, not RAR. This
+    is only configurable as a compat opt-out in case skipping the slow RAR code breaks something.
+  -->
+  <PropertyGroup Condition="'$(FindPackageDependenciesInResolveAssemblyReferences)' == ''">
+    <FindPackageDependenciesInResolveAssemblyReferences>false</FindPackageDependenciesInResolveAssemblyReferences>
+  </PropertyGroup>
+
+  <!--
+    Update static Reference items with NuGetPackageId metadata to set FindDependencies appropriately.
+    NetStandard.Library adds its references this way instead of the normal ResolvePackageDependencies,
+    and the mechanism is intended to be useable by other packages that have special logic for choosing 
+    assets
+   -->
+  <ItemGroup>
+    <Reference Update="@(Reference)">
+      <FindDependencies Condition="'%(NuGetPackageId)' != ''">$(FindPackageDependenciesInResolveAssemblyReferences)</FindDependencies>
```

Comment:
Shouldn't this also check for the `PackageName` metadata?

Reply:
No, this is specifically for the netstandard libs that only include NuGetPackageId.

Reply:
NETStandard.Library is special because it adds references in targets files rather than being raised from the lock file in the usual way.

For the regular case, we set FindDependencies here: https://github.com/dotnet/sdk/pull/1725/files#diff-4a26943c6456c3afd827c7b87e69d390R455 and there is no coupling to other metadata.